MatrixSDK/LocationSharing/MXLocationService.swift

```diff
@@ -0,0 +1,146 @@
+// 
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// MXLocationService handles live location sharing
+/// Note: Cannot use a protocol because of Objective-C compatibility
+@objcMembers
+public class MXLocationService: NSObject {
+    
+    // MARK: - Properties
+    
+    private let session: MXSession
```

Comment:
As the session holds a reference to `MXLocationService` instance, this line may cause a memory leak for the session object. We may have a weak/unowned variable here, like we did in most of other services.